core/src/test/java/org/jdbi/v3/locator/TestClasspathSqlLocator.java

```diff
@@ -0,0 +1,125 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.locator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.jdbi.v3.locator.ClasspathSqlLocator.findSqlOnClasspath;
+import static org.junit.Assert.assertEquals;
+
+import java.io.InputStream;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.jdbi.v3.H2DatabaseRule;
+import org.jdbi.v3.Handle;
+import org.jdbi.v3.exception.StatementException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class TestClasspathSqlLocator {
+    @Rule
+    public H2DatabaseRule db = new H2DatabaseRule();
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Before
+    public void setUp() {
+    }
+
+    @Test
+    public void testLocateNamed() throws Exception {
+        Handle h = db.openHandle();
+        h.insert(findSqlOnClasspath("insert-keith"));
+        assertEquals(1, h.select("select name from something").size());
+    }
+
+    @Test
+    public void testCommentsInExternalSql() throws Exception {
+        Handle h = db.openHandle();
+        h.insert(findSqlOnClasspath("insert-eric-with-comments"));
+        assertEquals(1, h.select("select name from something").size());
+    }
+
+    @Test
+    public void testNamedPositionalNamedParamsInPrepared() throws Exception {
+        Handle h = db.openHandle();
+        h.insert(findSqlOnClasspath("insert-id-name"), 3, "Tip");
+        assertEquals(1, h.select("select name from something").size());
+    }
+
+    @Test
+    public void testNamedParamsInExternal() throws Exception {
+        Handle h = db.openHandle();
+        h.createStatement(findSqlOnClasspath("insert-id-name"))
+                .bind("id", 1)
+                .bind("name", "Tip")
+                .execute();
+        assertEquals(1, h.select("select name from something").size());
+    }
+
+    @Test
+    public void testUsefulExceptionForBackTracing() throws Exception {
+        Handle h = db.openHandle();
+
+        exception.expect(StatementException.class);
+        exception.expectMessage("insert into something(id, name) values (:id, :name)");
+        exception.expectMessage("insert into something(id, name) values (?, ?)");
+        h.createStatement(findSqlOnClasspath("insert-id-name"))
+                .bind("id", 1)
+                .execute();
+    }
+
+    @Test
+    public void testNonExistentResource() throws Exception {
+        exception.expect(IllegalArgumentException.class);
+        findSqlOnClasspath("this-does-not-exist");
+    }
+
+    @Test
+    public void testCachesResultAfterFirstLookup() throws Exception {
+        ClassLoader ctx_loader = Thread.currentThread().getContextClassLoader();
```

Comment:
Yeah, some of these tests are just copy pasted from the old StatementLocator stuff--I will clean this up